bindings/pybind11/idyntree_core.cpp

```diff
@@ -91,58 +55,24 @@ void createFixSizeVector(pybind11::module& module,
       });
 }
 
-void matrixDynSizeClassDefinition(py::class_<MatrixDynSize>& matrix) {
-  matrix.def(py::init())
-      .def(py::init<unsigned, unsigned>())
+template <typename MatrixType>
+void baseMatrixDefinition(py::class_<MatrixType>& matrix) {
+  matrix
       .def("__getitem__",
-           [](MatrixDynSize& matrix,
+           [](MatrixType& matrix,
               std::pair<std::size_t, std::size_t> indices) {
              return matrix.getVal(indices.first, indices.second);
            })
       .def("__setitem__",
-           [](MatrixDynSize& matrix,
+           [](MatrixType& matrix,
               std::pair<std::size_t, std::size_t> indices, double item) {
              return matrix.setVal(indices.first, indices.second, item);
            })
-      .def("rows", &MatrixDynSize::rows)
-      .def("cols", &MatrixDynSize::cols)
-      .def("set_zero", &MatrixDynSize::zero)
-      .def("resize", &MatrixDynSize::resize)
-      .def("__repr__", &MatrixDynSize::toString)
-      .def_buffer([](MatrixDynSize& m) -> py::buffer_info {
-        return py::buffer_info(
-            m.data(),                                /* Pointer to buffer */
-            sizeof(double),                          /* Size of one scalar */
-            py::format_descriptor<double>::format(), /* Python struct-style
-                                                        format descriptor */
-            2,                                       /* Number of dimensions */
-            {m.rows(), m.cols()},                    /* Buffer dimensions */
-            {sizeof(double) * m.cols(), /* Strides (in bytes) for each index */
-             sizeof(double)});
-      });
-}
-
-template <unsigned rows, unsigned cols>
-void createFixSizeMatrix(pybind11::module& module,
-                         const std::string& class_name) {
-  py::class_<MatrixFixSize<rows, cols>>(module, class_name.c_str(),
-                                        py::buffer_protocol())
-      .def(py::init())
-      .def("__getitem__",
-           [](MatrixFixSize<rows, cols>& matrix,
-              std::pair<std::size_t, std::size_t> indices) {
-             return matrix.getVal(indices.first, indices.second);
-           })
-      .def("__setitem__",
-           [](MatrixFixSize<rows, cols>& matrix,
-              std::pair<std::size_t, std::size_t> indices, double item) {
-             return matrix.setVal(indices.first, indices.second, item);
-           })
-      .def("rows", &MatrixFixSize<rows, cols>::rows)
-      .def("cols", &MatrixFixSize<rows, cols>::cols)
-      .def("set_zero", &MatrixFixSize<rows, cols>::zero)
-      .def("__repr__", &MatrixFixSize<rows, cols>::toString)
-      .def_buffer([](MatrixFixSize<rows, cols>& m) -> py::buffer_info {
+      .def("rows", &MatrixType::rows)
+      .def("cols", &MatrixType::cols)
```

Comment:
Maybe these 2 are no longer needed now.
All types have explicit size given their type, e.g. a `Position` is size 3, etc

Reply:
This is done because in the c++ implementation it is possible to get the size of explicit size defined vector

Reply:
I think for users is convenient to have the same methods in C++ and Python, I do not think there are a lot of risks or problems related to have `rows`  and `cols`  methods in Python as well.

Reply:
I think it is ok (i.e. not dangerous) to keek `cols` and `rows` also in Python